test/integration/upload_test.lua

```diff
@@ -0,0 +1,161 @@
+local t = require('luatest')
+local g = t.group()
+
+local fio = require('fio')
+local fiber = require('fiber')
+local helpers = require('test.helper')
+
+g.before_all(function()
+    g.datadir = fio.tempdir()
+    g.upload_path_1 = fio.pathjoin(g.datadir, 'upload-1')
+    g.upload_path_2 = fio.pathjoin(g.datadir, 'upload-2')
+
+    g.cluster = helpers.Cluster:new({
+        datadir = g.datadir,
+        use_vshard = false,
+        server_command = helpers.entrypoint('srv_basic'),
+        cookie = helpers.random_cookie(),
+        replicasets = {{
+            alias = 'main',
+            uuid = helpers.uuid('a'),
+            roles = {},
+            servers = {
+                -- first upload group: s1 and s2
+                {env = {TARANTOOL_UPLOAD_PREFIX = g.upload_path_1}},
+                {env = {TARANTOOL_UPLOAD_PREFIX = g.upload_path_1}},
```

Comment:
Maybe it's reasonable to increase number of servers up to 5 or even more. In case of problems with concurrency and shared resource problems we have more chances to catch a problem in CI.